src/pytorch_lightning/tuner/lr_finder.py

```diff
@@ -265,6 +265,7 @@ def lr_find(
     # Restore initial state of model
     trainer._checkpoint_connector.restore(ckpt_path)
     trainer.strategy.remove_checkpoint(ckpt_path)
+    trainer.fit_loop.restarting = False  # reset restarting flag as checkpoint restoring sets it to True
```

Comment:
Since support for validate/test/predict was added, you need to do it for the root loop instance used. This will only work for `trainer.fit`

Reply:
I don't get this. What would an lr-finder do for validation/testing/prediction? The entire lr-find implementation relies on the fit loop only (which seems fine to me). All the other attributes should be reset by https://github.com/Lightning-AI/lightning/pull/15620/files#diff-246575a4bbb4b22771e909074e17fb462fabd8a971cd87c11aa56c6074da9d73L250 and https://github.com/Lightning-AI/lightning/pull/15620/files#diff-246575a4bbb4b22771e909074e17fb462fabd8a971cd87c11aa56c6074da9d73R266

Reply:
@justusschock is this addressed? this is the last pending item to get this merge... :otter:

Reply:
Like I said, I don't get the issue @carmocca mentioned and he didn't get back here...

Reply:
let's assume it resolved and you can sync and clarify offline so eventually implement in following PR :rabbit: